iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/LocationShifterTest.java

```diff
@@ -168,6 +174,34 @@ void shouldShiftLocationsWithoutExplicitNumbersWithRanges() {
     assertThat(shiftedRange).hasRange(1, 0, 3, 4);
   }
 
+  @Test
+  void shouldShiftYamlExceptions() {
+    setLinesSizes(ctx, 2);
+    shifter.addShiftedLine(ctx, 2, 1, 1);
+    var exception = new ParserException(null, Optional.of(Mockito.mock(Mark.class)), null,
+      Optional.of(new Mark("test", 1, 2, 1, new int[] {1, 1, 1, 1, 1}, 1)));
+
+    var shiftedException = shifter.shiftMarkedYamlException(ctx, exception);
+
+    Assertions.assertThat(shiftedException).isInstanceOf(ShiftedMarkedYamlEngineException.class);
+    Assertions.assertThat(shiftedException.getProblemMark()).isPresent()
+      .hasValueSatisfying(mark -> {
+        Assertions.assertThat(mark.getLine()).isEqualTo(1);
+        Assertions.assertThat(mark.getColumn()).isZero();
+      });
+  }
+
+  @Test
+  void shouldKeepYamlExceptionsWithoutMark() {
+    setLinesSizes(ctx, 2);
+    shifter.addShiftedLine(ctx, 2, 1, 1);
+    var exception = new ParserException(null, Optional.of(Mockito.mock(Mark.class)), null, Optional.empty());
+
+    var shiftedException = shifter.shiftMarkedYamlException(ctx, exception);
+
+    Assertions.assertThat(shiftedException).isSameAs(exception);
+  }
+
```

Comment:
I would like to see in at least one test how the logger error message looks like.

Reply:
Added a test with logging of shifting and discovered some problems. I reworked the recalculation of exception location to only use a single point (`problemMark.line`) instead of a whole range covering the whole snippet. We don't use this information later when converting to `ParseException`